src/AppBundle/Entity/User.php

```diff
@@ -163,4 +163,28 @@ public function eraseCredentials()
         // if you had a plainPassword property, you'd nullify it here
         // $this->plainPassword = null;
     }
+
+    /** @see \Serializable::serialize() */
+    public function serialize()
+    {
+        return serialize([
+            $this->id,
+            $this->username,
+            $this->password,
```

Comment:
I have to agree with you, it's quite unclear. I don't know what to do either maybe someone could enlight us on this particular topic ?